clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManager.java

```diff
@@ -414,8 +413,8 @@ public CompletableFuture<Void> signalPartitionsRevoked(Set<TopicPartition> parti
     private void logPausedPartitionsBeingRevoked(Set<TopicPartition> partitionsToRevoke) {
         Set<TopicPartition> revokePausedPartitions = subscriptions.pausedPartitions();
         revokePausedPartitions.retainAll(partitionsToRevoke);
-        if (!revokePausedPartitions.isEmpty()) {
-            log.info("The pause flag in partitions [{}] will be removed due to revocation.", revokePausedPartitions.stream().map(TopicPartition::toString).collect(Collectors.joining(", ")));
+        if (!revokePausedPartitions.isEmpty() && log.isInfoEnabled()) {
+            log.info("The pause flag in partitions [{}] will be removed due to revocation.", Utils.topicPartitionString(revokePausedPartitions));
```

Comment:
Given that er end up printing the string in between [], can't we simply print the set directly?
```suggestion
            log.info("The pause flag in partitions {} will be removed due to revocation.", revokePausedPartitions);
```

Reply:
You're right. Changed.